### ISSUE TITLE
Update concepts.py - age, gender, vulnerability

### ISSUE THREAD

**Issue by Charlotte-Morgan** [inasafe/inasafe#4092]
update concepts and icons to support analysis and reporting of age and gender 
#4069 
Age in 5 classes: infant, child, youth, adult, elderly
Vulnerability in 3 classes: under 5, over 60, disabled
Gender in 3 classes: female, child bearing age, pregnant or lactating

**Comment by ismailsunni** [comment]
@Charlotte-Morgan no male in gender?

And IMHO, your gender class is more like female's class (since all of the class are only apply for female).

Btw, I have put the placeholder in concepts.py that you can fill with the concept defintion (in my PR).

**Comment by samnawi** [comment]
Could we also include the sources of the definition and the grouping process in here?

**Comment by timlinux** [comment]
@samnawi yes I think we can use the standard citation system for this.

**Comment by samnawi** [comment]
Hi @timlinux I try to looking for some definition from what you ask in the last sprint meeting.

As the suggection from @ismailsunni, I try to take some definition from [here](https://github.com/inasafe/inasafe/blob/develop/safe/definitions/concepts.py#L566) then find the empty list also update some sentence from the existing.

To flag the sentence that I wrote, formatted into italic.

# Age Group
## 3 classes
**Youth**
	A person aged between 0 and 14 years.
**Adult**
	Person aged between 15 and 64 years, usually of working age.
**Elderly**
	Person aged 64 years and over.

## 5 class
**Infant**
	A very young child or baby.
**Child**
	A young human being below the age of puberty or below the legal age of majority.
_**Youth**_
	_The period between childhood and adult age (https://en.oxforddictionaries.com/definition/youth) ._ 
_**Adult**_
	_A person who has reached the age of majority. (https://en.oxforddictionaries.com/definition/adult)._
**_Elderly_**
	_Having lived for a long time; no longer young. (https://en.oxforddictionaries.com/definition/old)._

# Gender Group
**Male**
	Relating to the characteristics of men.
**Female**
	Relating to the characteristics of women.
**_Childbearing_** 
	_The female group in the reproductive age between 15 - 19 years, unmerried, merried or widow (Indonesia Health Ministry - http://www.depkes.go.id/resources/download/pusdatin/lain-lain/Data%20dan%20Informasi%20Kesehatan%20Profil%20Kesehatan%20Indonesia%202016%20-%20%20smaller%20size%20-%20web.pdf)_
**_Non Childbearing_**
	_Beyond the childbearing age_
**Pregnant**
A female having a child developing in the uterus.
**_Lactating_**
_(of a female mammal) secrete milk. (https://en.oxforddictionaries.com/definition/lactate)._

# Vulnerability Group
**_Under 5 years_**
	Anak di bawah lima tahun (balita) adalah anak laki-laki atau perempuan yang berumur 0-59 bulan 
	Under-Five Children are children aged between 0 and 59 months.(https://www.bps.go.id/index.php/istilah/index?Istilah_page=3)_
**_More than 60 years_**
	 _Older persons aged 60 years or over (http://www.un.org/en/development/desa/population/publications/pdf/ageing/WorldPopulationAgeing2013.pdf)_
**_Disability_**
	_Physical defect refers to total and permanent disablement which causes a person to be unable to carry out work which would provide an income which might reasonably be obtained considering their education, expertise, skill and experience.(https://www.bps.go.id/index.php/istilah/index?Istilah_page=10)._

**Comment by Charlotte-Morgan** [comment]
@samnawi - can we please get the default rates for these concepts too please

**Comment by Charlotte-Morgan** [comment]
Quick update from @samnawi and I on this work.
We are reviewing age, gender and vulnerability concepts / concept_groups to align with humanitarian standards.  thinking is being documented here: https://docs.google.com/spreadsheets/d/1NFGIRPyb69yO4I-8QkB4MaiGn-Te2L0gPd9yDaj6tiA/edit#gid=0

Steps to progress this:
- [ ] get icons for all concepts
- [ ] write definitions for all concepts
- [ ] set defaults for age, gender concepts
- [ ] revise concept groups and tweak words

**Comment by ismailsunni** [comment]
Not related directly, but I also had an idea to put the icon in the defintions like I mentioned in https://github.com/inasafe/inasafe/issues/3470

+1 for the icon.

**Comment by Charlotte-Morgan** [comment]
Yes - i was just looking at my notes on how to do this.
Add icon files to the resources
Add entries to definitions.lic

I will also be working on concepts.py so if you can give me a sample line of code to add to the standard concept boiler plate, I can easily list the icon as you were thinking in #3470.

I think i discussed this with @timlinux before and recall for some reason we decided not to do it. Probably not a good reason or I should remember

**Comment by Charlotte-Morgan** [comment]
here is the existing concept snip
```
concepts['adult'] = {
    'group': tr('Demographics'),
    'key': 'adult',
    'description': tr(
        'Person aged between 15 and 64 years, usually of working age.'),
    'citations': [
        {
            'text': tr(
                'CIA (2016)The World Factbook.'),
            'link': u'https://www.cia.gov/library/publications/'
                    u'resources/the-world-factbook/'
        }
    ],
}
```

**Comment by samnawi** [comment]
Please check this link
https://docs.google.com/spreadsheets/d/1NFGIRPyb69yO4I-8QkB4MaiGn-Te2L0gPd9yDaj6tiA/edit?usp=sharing

**Comment by Charlotte-Morgan** [comment]
@ismailsunni - following discussion with @timlinux and some reworking with @samnawi we would like to make some changes to the age concepts and group. Here is the proposal. 
* We have 3 default classes as per UN and other international humanitarian standards.
* Youth, adult and elderly
* we allow 4 classes: infant, child, adult and elderly.

![image](https://cloud.githubusercontent.com/assets/5743135/25892771/ae90c0a6-359f-11e7-8e91-82088bb99376.png)

This means that if a person uses youth, they should not use child or infant (plus the inverse)
It should not be possible to use any combination of child and youth or infant and youth.
These concepts and ratios are all in PR #4144 if you want to consider this proposal.

@felix-yew - can you please find an ice cream icon for @ismailsunni ?

**Comment by Charlotte-Morgan** [comment]
@ismailsunni 

Further to the above. We would like to take your advice and revise the gender group to include males and females.  We propose that all other concepts such as pregnant and lactating and child bearing age, belong elsewhere.
@felix-yew  - could you please make that 2 ice cream icons?

**Comment by Charlotte-Morgan** [comment]
@ismailsunni 

Finally: we propose that we modify the vulnerability groups as follows:
Vulnerability age: under 5, over 60
Vulnerability gender: female, child bearing, pregnant and lactating
Vulnerability disability: disabled

CMIIW - this proposal uses the same concepts that are already define but changes some groups and adds more groups.
Is this something you are able to support or would you like me to attempt to implement this?
I think after PR #4144 is merged.

@felix-yew - do you have an icon for an ice cream van?

**Comment by timlinux** [comment]
> Vulnerability age: under 5, over 60
> Vulnerability gender: female, child bearing, pregnant and lactating
> Vulnerability disability: disabled

So we take the one existing group and split it into 3 separate groups? Sounds fine to me

**Comment by timlinux** [comment]
> Further to the above. We would like to take your advice and revise the gender group to include males and females. We propose that all other concepts such as pregnant and lactating and child bearing age, belong elsewhere.

I think this is fine too - we can make two groups:

**Gender group**: Males, Females
**Women group**: Pregnant, Lactating, Child Bearing Age

**Comment by timlinux** [comment]
> ![image](https://cloud.githubusercontent.com/assets/5743135/25892771/ae90c0a6-359f-11e7-8e91-82088bb99376.png)

> This means that if a person uses youth, they should not use child or infant (plus the inverse)
It should not be possible to use any combination of child and youth or infant and youth.
These concepts and ratios are all in PR #4144 if you want to consider this proposal.

This one might be a bit more trick. Basically it implies another level in the heirachy:

Before:

![screen shot 2017-05-10 at 1 15 42 pm](https://cloud.githubusercontent.com/assets/178003/25896050/da5e66b0-3582-11e7-9bdc-2694c9efa2c0.png)

After:

![screen shot 2017-05-10 at 1 16 33 pm](https://cloud.githubusercontent.com/assets/178003/25896062/ed4a460e-3582-11e7-95e3-b311385ddcef.png)

Or (alternative after):

![screen shot 2017-05-10 at 1 22 33 pm](https://cloud.githubusercontent.com/assets/178003/25896893/342e24f2-3586-11e7-8116-5368b1dd29ec.png)


It also implies that reporting will be aware of subgroups and do the right thing with them automatically. I think we need to think about this a little to figure out what / how it will be done.

**Comment by Charlotte-Morgan** [comment]
I don't think we need the women group. I've put them as a vulnerability group

**Comment by Charlotte-Morgan** [comment]
Ok. Let's stick with three only for now and tuck infants and children away

**Comment by Charlotte-Morgan** [comment]
@timlinux, @ismailsunni , @samnawi 
thanks for the feedback and discussion around the demographic group. The more I think - maybe we need to separate the data handling and analysis from the reporting.
For the demography group - is it another level in the heirarchy or a group?


Separating data from reporting:
I think we should allow users to define all the concepts for infant, child, youth, adult and elderly.
At present the tool does not allow the same field to be assigned to to more than one class within a group. A simple solution may be to add a group rather than the  heirarchy as suggested by @timlinux 
ie turn age group into age group 3 and age group 4 - just like we have different hazard classifications.
This would then be a new tab in the tools. 
I think this is a short term solution - maybe there is a better way in the longer term

When it comes to reporting; then it would be sensible to make infant and child mutually exclusive from youth as per the diagram. 
https://cloud.githubusercontent.com/assets/5743135/25892771/ae90c0a6-359f-11e7-8e91-82088bb99376.png
so a report would have adult youth elderly or infant child adult elderly
the sum of infant, child, youth, adult and elderly would be greater than 1

**Comment by ismailsunni** [comment]
Hi @Charlotte-Morgan 
> I think we should allow users to define all the concepts for infant, child, youth, adult and elderly.
At present the tool does not allow the same field to be assigned to to more than one class within a group. A simple solution may be to add a group rather than the heirarchy as suggested by @timlinux
ie turn age group into age group 3 and age group 4 - just like we have different hazard classifications.
This would then be a new tab in the tools.
I think this is a short term solution - maybe there is a better way in the longer term

By doing this, what will happen if the user fill both the age group 3 and group 4? Which one that we want to use? Both of group have `elderly` and `adult`, should they have the same input? If not, we need to differentiate them in term of storing in the metadata. Next problem which one that we want to use when we do the analysis?

For the reporting part, how can the reporting knows that youth = child + infant? I am not sure we want to hard code it (cmmiw @timlinux )

**Comment by Charlotte-Morgan** [comment]
hi @ismailsunni 

> By doing this, what will happen if the user fill both the age group 3 and group 4? Which one that we want to use? Both of group have elderly and adult, should they have the same input? If not, we need to differentiate them in term of storing in the metadata. Next problem which one that we want to use when we do the analysis?

For the reporting part, how can the reporting knows that youth = child + infant? I am not sure we want to hard code it (cmmiw @timlinux )

All good points. Maybe we should keep it simple for now and use a default of 3. We can keep the others but maybe disable infant and child for now.
When @samnawi reviewed the way different organisations work with age groups most use only 3 groups. BNPB would like more but do not have a consistent requirement. Which is why I suggest we keep it simple for now and think more about how to accommodate these overlapping issues.

If you keep it as a single group - ie tab for input; could you put in a box to ask users which to report on? - just another idea :)  - or this goes back to a users options setting

**Comment by felix-yew** [comment]
Ok @Charlotte-Morgan - this is a lot of ice cream for mas @ismailsunni 
![noun_746208_cc](https://cloud.githubusercontent.com/assets/13621886/25938574/e753c3ba-3659-11e7-8c1b-9ee13edbc66d.png)

with the ice cream truck
![noun_1033650_cc](https://cloud.githubusercontent.com/assets/13621886/25938595/f5fdb33a-3659-11e7-8b24-e44b6ba2f65f.png)

Please enjoy! 😄

**Comment by Charlotte-Morgan** [comment]
thank you mas @felix-yew

**Comment by ismailsunni** [comment]
Thanks for the icon @felix-yew  @Charlotte-Morgan , I will create new ticket for the real one next week :D

**Comment by samnawi** [comment]
I've already tested the calculation as what @Charlotte-Morgan asking on tuesday morning discussion about the gender group.

I think this is work as what we want. 
![screenshot_10](https://cloud.githubusercontent.com/assets/8871958/26146432/5c1adee4-3b1a-11e7-9b47-d05fc3a345d6.jpg)

From the picture in above, we can see that as long the ratio is adjusted to the appropriate number (based on the total pop (male and female) or just total pop (just female), InaSAFE will generate the same result. 

This is the keyword that I set up for the population data
![image](https://cloud.githubusercontent.com/assets/8871958/26146811/ae0ee316-3b1b-11e7-9541-fedfb9ac7083.png)
